python/astro_metadata_translator/translators/decam.py

```diff
@@ -248,3 +253,41 @@ def to_detector_name(self):
         # Docstring will be inherited. Property defined in properties.py
         name = self.to_detector_unique_name()
         return name[1:]
+
+    @classmethod
+    def fix_header(cls, header):
```

Comment:
The new interface for fix_header is 

```python
fix_header(header, instrument, obsid, filename=None)
```

you can then build the log message prefix with

```python
        # Calculate the standard label to use for log messages
        log_label = cls._construct_log_prefix(obsid, filename)
```